scripts/us_bjs/nps/preprocess_data.py

```diff
@@ -0,0 +1,222 @@
+    df["Count_Custody_NotAUSCitizen"] = df["NCITZTOTF"] + df["NCITZTOTM"]
+    df["Count_Custody_NotAUSCitizen_MaxSentenceGreaterThan1Year_Female"] = df["NCITZGT1F"]
+    df["Count_Custody_NotAUSCitizen_MaxSentenceGreaterThan1Year_Male"] = df["NCITZGT1M"]
+    df["Count_Custody_NotAUSCitizen_MaxSentenceGreaterThan1Year"] = df["NCITZGT1F"] + df["NCITZGT1M"]
+    df["Count_Custody_NotAUSCitizen_MaxSentence1YearOrLess_Female"] = df["NCITZLE1F"]
+    df["Count_Custody_NotAUSCitizen_MaxSentence1YearOrLess_Male"] = df["NCITZLE1M"]
+    df["Count_Custody_NotAUSCitizen_MaxSentence1YearOrLess"] = df["NCITZLE1F"] + df["NCITZLE1M"]
+    df["Count_Custody_NotAUSCitizen_Unsentenced_Female"] = df["NCITZUNSF"]
+    df["Count_Custody_NotAUSCitizen_Unsentenced_Male"] = df["NCITZUNSM"]
+    df["Count_Custody_NotAUSCitizen_Unsentenced"] = df["NCITZUNSF"] + df["NCITZUNSM"]
+    df["Count_Custody_Under18_Female"] = df["CUSLT18F"]
+    df["Count_Custody_Under18_Male"] = df["CUSLT18M"]
+    df["Count_Custody_Under18"] = df["CUSLT18F"] + df["CUSLT18M"]
+    get_total_jurisdiction_columns(df)
+    
+def convert_geoId(fips_code):
+    """Creates geoId column"""
+    return 'geoId/' + str(fips_code).zfill(2)
+
+def convert_missing_value_to_nan(value):
+    """codes for missing values are always negative and actual data is always >= 0"""
+    if isinstance(value, int) and value < 0:
+        return float("nan")
+    else:
+        return value
+        
+def convert_nan_to_empty_cell(value):
+    if pd.isna(value):
+        return ''
+    else:
+        return value
+    
+def preprocess_df(raw_df):
+    """cleans raw_df
+
+    Args:
+        raw_data: raw data frame to be used as starting point for cleaning
+    """
+    df = raw_df.copy()
+    column_names = list(df.columns)
+    df['GeoId'] = df['STATEID'].apply(convert_geoId)
+    
+    # convert missing values to NaN for aggregation
+    for column_name in column_names:
+        df[column_name] = df[column_name].apply(convert_missing_value_to_nan)
+    
+    #get columns matching stat var names and add aggregate columns
+    get_columns(df)
+    
+    #convert NaN to empty cell
+    for column_name in column_names:
+        df[column_name] = df[column_name].apply(convert_nan_to_empty_cell)
+        
+    #filter out unnecessary columns
+    return df[DESIRED_COLUMNS]
+
+def main(args):
+        filename = args.input_file
+        print('Processing {0}'.format(filename))
+        df = pd.read_csv(filename, delimiter='\t')
+        processed_df = preprocess_df(df)
+        processed_df.to_csv(args.input_file.replace('.tsv', '_processed.csv'),
+                            index=False)
+        print('Done processing {0}'.format(args.input_file))
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description="NPS Processing Script")
+    parser.add_argument('-i',
+                        dest='input_file',
+                        default='NPS_1978-2018_Data.tsv')
+    args = parser.parse_args()
```

Comment:
Could we use absl.flags instead? so much nicer. e.g.: https://github.com/datacommonsorg/data/blob/96bf0aeb9ff1ba401ebeca6d9d777f7c1c3d4f7e/scripts/datacommons/place_cohorts/generate_cohort_set.py#L19-L27